core/iwasm/libraries/lib-wasi-threads/stress_test/mutex_common.h

```diff
@@ -0,0 +1,261 @@
+/*
+ * Copyright (C) 2023 Amazon.com Inc. or its affiliates. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ */
+
+#ifndef MUTEX_COMMON_H
+#define MUTEX_COMMON_H
+
+#include <pthread.h>
+#include <stdio.h>
+#include <assert.h>
+#include <errno.h>
+#include <unistd.h>
+#include <stdbool.h>
+
+typedef struct {
+    pthread_mutex_t *mutex;
+    int *counter;
+    int limit;
+    bool is_sleeping;
+} MutexCounter;
+
+enum constants {
+    NUM_ITER = 100000,
+    NUM_THREADS = 12,
+    NUM_RETRY = 8,
+    RETRY_SLEEP_TIME_US = 1000,
+};
+
+// This enum defines whether thread should sleep to increase contention
+enum sleep_state {
+    NON_SLEEP = 0,
+    SLEEP = 1,
+};
+
+void
```

Comment:
*This is a comment for all the functions / global variables defined in the file*

I'd strongly advise to move the implementation to .c file or make them static. Otherwise, if the file is included in two compilation units and they're linked together, there'll be a linker error (due to multiple definitions). I know we don't have this scenario now, but with little effort we can make this better for the potential future.